### PR TITLE
feat(ibm-no-operation-requestbody): add new spectral rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces a new spectral-style rule
'ibm-no-operation-requestbody' which will check to make sure that certain operations (delete, get, head, options) do not have a requestBody defined (as recommended by the API Handbook).

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

### CODE REVIEW

File: docs/ibm-cloud-rules.md [hunk 1]
```diff
@@ -2980,6 +2991,103 @@ paths:
...
+</pre>
+</td>
+</tr>
+<tr>
+<td valign=top><b>Compliant example:</b></td>
+<td>
+<pre>
+paths:
+  '/v1/things/search':
+    post:     &lt;&lt;&lt; GET changed to POST
```
Comment (dpopp07) [inline docs/ibm-cloud-rules.md:3070]: Perhaps this should just be a GET without a request body? Not a big deal, I get what you are demonstrating here, just seemed odd to me that the operation switched

File: packages/ruleset/src/functions/delete-body.js [hunk 2]
```diff
@@ -12,6 +12,22 @@ module.exports = function (operation, _opts, context) {
...
+    // but spectral will still invoke the rule despite the severity setting.
+    // If a user enables this rule in their custom ruleset, we'll want to log a warning
+    // to alert them to the deprecation status.
+    if (context.rule && context.rule.definition) {
+      const severity = context.rule.definition.severity;
+      if (
+        (typeof severity === 'number' && severity >= 0) ||
+        (typeof severity === 'string' && severity !== 'off')
+      ) {
+        logger.warn(
```
Comment (dpopp07) [inline packages/ruleset/src/functions/delete-body.js:26]: Will this log more than once if the rule is enabled and there are multiple operations?

File: packages/ruleset/src/functions/index.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ module.exports = {
   mergePatchProperties: require('./merge-patch-properties'),
   operationIdCasingConvention: require('./operationid-casing-convention'),
   operationIdNamingConvention: require('./operationid-naming-convention'),
+  operationRequestBody: require('./operation-requestbody'),
```
Comment (dpopp07) [inline packages/ruleset/src/functions/index.js:27]: I know this is picky 😅 but to be consistent with my comment from last time - it really helps me when debugging rules and digging through code to have the rule/function files match the rule ID (i.e. `no-operation-requestbody`) when possible. If it's a hassle, I don't mind leaving it but my guess is it'll be less intuitive down the road than it seems now

**Review by dpopp07 — COMMENTED** [review]
Looks good! A couple minor things but the logic itself looks great

File: docs/ibm-cloud-rules.md (hunk 1 above)
Comment (padamstx) [reply docs/ibm-cloud-rules.md:3070]: haha, yeah I guess I was looking at it from the opposite angle... i.e. using a GET where we should really use a POST instead :)
I'll change it

File: packages/ruleset/src/functions/delete-body.js (hunk 2 above)
Comment (padamstx) [reply packages/ruleset/src/functions/delete-body.js:26]: No, this is inside the "if (!logger)" above, which we'll enter only the first time we are invoked.  I initially started setting up a separate flag to make sure we log the warning only once, then realized I could piggyback off the if block that was already there to initialize the logger.

Comment (dpopp07) [reply packages/ruleset/src/functions/delete-body.js:26]: Ah, of course - I figured you wouldn't have implemented it in a way that would log multiple messages but I was struggling to understand how you did it (totally missed that context 🙂)

File: packages/ruleset/src/functions/index.js (hunk 3 above)
Comment (padamstx) [reply packages/ruleset/src/functions/index.js:27]: actually, I had initially was going to use "ibm-operation-requestbody" as the rule name, but then changed to "ibm-no-operation-requestbody" and forgot to rename the files after that.  will fix

Comment (padamstx) [reply packages/ruleset/src/functions/index.js:27]: done

**Comment by padamstx** [comment]
Ok, just pushed changes to address review comments

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**